packages/nextjs/src/app/utils.ts

```diff
@@ -1,11 +1,11 @@
 // Copyright © 2024 Ory Corp
 // SPDX-License-Identifier: Apache-2.0
+import { FlowType, OnRedirectHandler } from "@ory/client-fetch"
 import { headers } from "next/headers"
-import { redirect } from "next/navigation"
-import { OnRedirectHandler } from "@ory/client-fetch"
+import { redirect, RedirectType } from "next/navigation"
 
+import { urlQueryToSearchParams } from "next/dist/shared/lib/router/utils/querystring"
```

Comment:
This seems like it might break in different versions of next js. How do we make sure that the library doesn't only work with a specific version of it?

Reply:
We could copy over the function: https://github.com/vercel/next.js/blob/dbd5e1b274d30f9107141475eba116a8118bc346/packages/next/src/shared/lib/router/utils/querystring.ts#L35-L47

Reply:
yeah ok